### ISSUE TITLE
Response from Minions not reaching Master when message signing is turned on.

### ISSUE THREAD

**Issue by Church-** [saltstack/salt#52143]
### Description of Issue/Question
When turning on pub key signing for master and minions, the minion will execute the sent commands or apply sent states, however it fails to send a response back to the master correctly causing the master to return "Minion did not return. [No response]."
### Setup
  Master-config:
  master_sign_pubkey: True
  sign_pub_messages: True
  require_minion_sign_messages: True
  drop_messages_signature_fail: True

-------------------------------------------------------

  Minion-config:
  verify_master_pubkey_sign: True
  always_verify_signature: True
  master_sign_key_name: master_sign

### Steps to Reproduce Issue
Set the needed options in your master and minion configs and copy over the master_sign.pub key over from /etc/salt/pki/master/ on the master to /etc/salt/pki/minion/ on the minion before restarting both and trying to test a command.

### Versions Report
Salt Version:
           Salt: 2019.2.0
 
Dependency Versions:
           cffi: 1.12.1
       cherrypy: 2.3.0
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-43-generic
         system: Linux
        version: Ubuntu 18.04 bionic

**Comment by Ch3LL** [comment]
You need to set `minion_sign_messages: True` so your minions are signing messages since you are requiring this with: `require_minion_sign_messages: True`

I tested this with your settings and after adding minion_sign_messages it worked.

**Comment by Church-** [comment]
Yeah that fixed it fully, thanks! Guess I must've missed it while reading the pki tutorial.

**Comment by Gunni** [comment]
`minion_sign_messages` is missing from the default minion config file, maybe add it?

And or add a reference to the master config file that it needs to be set?

I had to google around until i found this ticket to get this working.

**Comment by Ch3LL** [comment]
I went ahead and re-opened this and will label as a Documentation bug so we can improve here. Thanks @Gunni
